packages/test-kit/src/with-feature.ts

```diff
@@ -255,43 +260,70 @@ export function withFeature(withFeatureOptions: IWithFeatureOptions = {}) {
                 throw new Error('Engine HTTP server is closed!');
             }
 
-            const runningFeature = await executableApp.runFeature({
-                featureName,
-                configName,
-                runtimeOptions: runOptions,
-                overrideConfig: config,
-            });
+            let runningFeature;
+            const runFeature = async () => {
+                runningFeature = await executableApp.runFeature({
+                    featureName,
+                    configName,
+                    runtimeOptions: runOptions,
+                    overrideConfig: config,
+                });
+            };
+
+            if (persist) {
+                before('load feature', runFeature);
```

Comment:
we're inside getLoadedFeature, which means we're inside an it, so we shouldn't call `before` here. hooking into before should be in withFeature closure (which is executed in describe).